src/iop/colorbalance.c

```diff
@@ -548,19 +1369,625 @@ static gboolean dt_iop_area_draw(GtkWidget *widget, cairo_t *cr, dt_iop_module_t
 }
 #endif
 
+static void autogrey_callback(GtkWidget *button, gpointer user_data)
+{
+  dt_iop_module_t *self = (dt_iop_module_t *)user_data;
+  if(self->dt->gui->reset) return;
+  
+  dt_iop_request_focus(self);
+  dt_control_queue_redraw();
+
+  if(self->request_color_pick == DT_REQUEST_COLORPICK_MODULE)
```

Comment:
The control flow is a bit complex and seems to be equivalent to:

```
if(self->request_color_pick == DT_REQUEST_COLORPICK_OFF)
{
    dt_iop_request_focus(self);
    self->request_color_pick = DT_REQUEST_COLORPICK_MODULE;
}
else
{
    dt_iop_colorbalance_params_t *p = (dt_iop_colorbalance_params_t *)self->params;
    dt_iop_colorbalance_gui_data_t *g = (dt_iop_colorbalance_gui_data_t *)self->gui_data;

    dt_dev_reprocess_all(self->dev);

    ...
}
```

Again, this must be done in all color picker related callbacks.